src/pyairios/__init__.py

```diff
@@ -77,21 +79,23 @@ async def fetch(self) -> AiriosData:
         """Get the data from all nodes at once."""
         data: dict[int, AiriosNodeData] = {}
 
-        brdg_data = await self.bridge.fetch_bridge()
-        if brdg_data["rf_address"] is None:
+        brdg_data = await self.bridge.fetch_bridge_data()
+        if brdg_data is None or brdg_data["rf_address"] is None:
             raise AiriosException("Failed to fetch node RF address")
         bridge_rf_address = brdg_data["rf_address"].value
         data[self.bridge.slave_id] = brdg_data
 
-        for node in await self.bridge.nodes():
-            if node.product_id == ProductId.VMD_02RPS78:
-                vmd = VMD02RPS78(node.slave_id, self.bridge.client)
-                vmd_data = await vmd.fetch_vmd_data()
-                data[node.slave_id] = vmd_data
-            if node.product_id == ProductId.VMN_05LM02:
-                vmn = VMN05LM02(node.slave_id, self.bridge.client)
-                vmn_data = await vmn.fetch_vmn_data()
-                data[node.slave_id] = vmn_data
+        prids = brdg_data["product_ids"]
+        if prids is not None and brdg_data["models"] is not None:
+            for _node in await self.bridge.nodes():
+                for key, _id in prids.items():
```

Comment:
This needs to be reversed. Instead of iterating the supported models first, iterate the bound nodes and then check if it is supported or not (iterating `prids`). The reason is that you can have several physical machines bound, all of them with the same product id but with its own virtual modbus device on the bridge.

Reply:
If reversed, it means awaiting every loop. As is, we await just once, and fetch a matching model (might be the same as in previous loop): same result